src/csharp/Microsoft.Spark.Extensions.Delta.UnitTest/DeltaTableTests.cs

```diff
@@ -0,0 +1,180 @@
+            ValidateTutorialDataFrame(Enumerable.Range(0, 5), data);
+
+            // Create a second iteration of the table.
+            data = _spark.Range(5, 10);
+            data.Write().Format("delta").Mode("overwrite").Save(path);
+
+            // Load the data into a DeltaTable object.
+            var deltaTable = DeltaTable.ForPath(path);
+
+            // Validate that deltaTable contains the the sequence [5 ... 9].
+            ValidateTutorialDataFrame(Enumerable.Range(5, 5), deltaTable.ToDF());
+
+            //// Update every even value by adding 100 to it.
+            deltaTable.Update(
+                condition: Functions.Expr("id % 2 == 0"),
+                set: new Dictionary<string, Column>()
+                {
+                    { "id", Functions.Expr("id + 100") }
+                });
+
+            //// Validate that deltaTable contains the the data:
+            //// +---+
+            //// | id|
+            //// +---+
+            //// |  5|
+            //// |  7|
+            //// |  9|
+            //// |106|
+            //// |108|
+            //// +---+
+            ValidateTutorialDataFrame(new List<int>() { 5, 7, 9, 106, 108 }, deltaTable.ToDF());
+
+            // Delete every even value.
+            deltaTable.Delete(condition: Functions.Expr("id % 2 == 0"));
+
+            // Validate that deltaTable contains:
+            // +---+
+            // | id|
+            // +---+
+            // |  5|
+            // |  7|
+            // |  9|
+            // +---+
+            ValidateTutorialDataFrame(new List<int>() { 5, 7, 9 }, deltaTable.ToDF());
+
+            // Upsert (merge) new data.
+            DataFrame newData = _spark.Range(0, 20).As("newData").ToDF();
+
+            deltaTable.As("oldData")
+                .Merge(newData, "oldData.id = newData.id")
+                .WhenMatched()
+                .Update(
+                    new Dictionary<string, Column>() {
+                                { "id", Functions.Col("newData.id") }
+                    })
+                .WhenNotMatched()
+                .InsertExpr(
+                    new Dictionary<string, string>() {
+                                { "id", "newData.id" }
+                    })
+                .Execute();
+
+            // Validate that the resulTable contains the the sequence [0 ... 19].
+            ValidateTutorialDataFrame(Enumerable.Range(0, 20), deltaTable.ToDF());
+        }
+
+        /// <summary>
+        /// Test that methods return the expected signature.
+        /// </summary>
+        [Fact]
+        public void TestSignatures()
+        {
+            using TemporaryDirectory tempDirectory = new TemporaryDirectory();
+            string path = Path.Combine(tempDirectory.Path, "delta-table");
+
+            DataFrame rangeRate = _spark.Range(15);
+            rangeRate.Write().Format("delta").Save(path);
+
+            DeltaTable table = DeltaTable.ForPath(path);
```

Comment:
what about
```csharp
ForPath(SparkSession sparkSession, string path) 
```

Let's verify that all the methods are called.

Reply:
Added